lib/protocol/protocol.js

```diff
@@ -329,18 +358,30 @@ function decodeMessageSet (topic, partition, messageSet, cb, maxTickMessages, hi
               message.timestamp = new Date(vars.timestamp);
             }
 
-            return cb(null, 'message', message);
+            return enqueue((next) => {
+              emit(null, 'message', message);
+              next(null);
+            });
           }
-          codec.decode(vars.value, function (error, inlineMessageSet) {
-            if (error) return; // Not sure where to report this
-            decodeMessageSet(topic, partition, inlineMessageSet, cb, maxTickMessages, highWaterOffset);
+          enqueue((next) => {
+            codec.decode(vars.value, function (error, inlineMessageSet) {
+              if (error) emit(error);
```

Comment:
If there's an error from the codec we should return instead of trying to decode?

Reply:
eek, yep.